code/game/objects/structures/reflector.dm

```diff
@@ -265,3 +259,54 @@
 		return
 	else
 		return ..()
+
+// tgui menu
+
+/obj/structure/reflector/ui_interact(mob/user, datum/tgui/ui)
+	if(!finished)
+		user.balloon_alert(user, "nothing to rotate!")
+		return
+	if(!can_rotate)
+		user.balloon_alert(user, "can't rotate!")
+		return
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Reflector")
+		ui.open()
+
+/obj/structure/reflector/attack_robot(mob/user)
+	ui_interact(user)
+	return
+
+/obj/structure/reflector/ui_state(mob/user)
+	return GLOB.physical_state //Prevents borgs from adjusting this at range
+
+/obj/structure/reflector/ui_data(mob/user)
+	var/list/data = list()
+	data["rotation_angle"] = rotation_angle
+	data["reflector_name"] = name
+
+	return data
+
+/obj/structure/reflector/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+	switch(action)
+		if("rotate")
+			if (!can_rotate || admin)
+				return FALSE
+			var/new_angle = text2num(params["rotation_angle"])
+			if(isnull(new_angle))
+				CRASH("A string was inputted to [src] instead of a number while rotating, somehow.")
```

Comment:
This isn't an error, this is an expected possibility resulted with TGUI exploitation. log_href_exploit is the correct logging for this.

Reply:
I got told by another maintainer to do CRASH() in the first review. The feedback is getting confusing.

I will change it to log href